test/e2e/e2e_test.go

```diff
@@ -182,4 +182,28 @@ var _ = Describe("Workload cluster creation", func() {
 			}, result)
 		})
 	})
+
+	Context("Creating a control-plane cluster with an internal load balancer", func() {
+		It("Should create a cluster with 1 control-plane and 1 worker node with an internal load balancer", func() {
```

Comment:
i think that is fine for now, but we need to check if the LB created is internal, maybe do something similar what capz does: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/3be87785773925374b601b3002c0c32edd5b08ec/test/e2e/azure_privatecluster.go#L47

thanks for adding more tests, this is cool

Reply:
Yes, using just the internal LB is a bit more complex as it will require adding a proxy or additional way to access the internal endpoint. I am currently working on a downstream test and will add an additional test here in the future. Thanks for the pointer to way capz does it, that's useful.